File: FibonacciSeriesUpToNthTerm/FibonacciSeriesUpToNthTerm.cpp
/*
https://www.geeksforgeeks.org/problems/fibonacci-series-up-to-nth-term/1?utm_source=geeksforgeeks&utm_medium=newui_home&utm_campaign=potd
*/

//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
    vector<int> Series(int n) {
        vector<int>res;
        int prev=0, curr=1, mod=1e9+7;
        res.push_back(prev);
        while(res.size()!=n+1){
            res.push_back(curr);
            int temp=curr;
            curr+=prev;
            curr%=mod;
            prev=temp;
        }
        return res;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution obj;

        vector<int> ans = obj.Series(n);
        for (auto x : ans) cout << x << " ";
        cout << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: FibonacciSeriesUpToNthTerm/FibonacciSeriesUpToNthTerm.java
/*
https://www.geeksforgeeks.org/problems/fibonacci-series-up-to-nth-term/1?utm_source=geeksforgeeks&utm_medium=newui_home&utm_campaign=potd
*/

//{ Driver Code Starts
import java.io.*;
import java.util.*;

class Main {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());
            Solution obj = new Solution();
            int ans[] = obj.Series(n);
            for (int i : ans) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends


class Solution {

    int[] Series(int n) {
        int[] res=new int[n+1];
        int idx=0, prev=0, curr=1, mod=1000000000+7;
        res[idx++]=prev;
        while(idx<=n){
            res[idx]=curr;
            int temp=curr;
            curr+=prev;
            curr%=mod;
            prev=temp;
            idx++;
        }
        return res;
    }
}